source/AST/ASTVisitor.cpp

```diff
@@ -65,40 +69,49 @@ ASTVisitor(
 // a relatively small amount of memory (vs storing
 // USRs directly).
 //
-SymbolID
+bool
 ASTVisitor::
-getSymbolID(
-    Decl const* D)
+extractSymbolID(
+    const Decl* D,
+    SymbolID& id)
 {
-    llvm::SmallString<128> USR;
-    if(index::generateUSRForDecl(D, USR))
-        return SymbolID::zero;
-    auto r = llvm::SHA1::hash(arrayRefFromStringRef(USR));
-    return SymbolID(r.data());
+    usr_.clear();
+    if(index::generateUSRForDecl(D, usr_))
+        return false;
+    id = SymbolID(llvm::SHA1::hash(
+        arrayRefFromStringRef(usr_)).data());
+    return true;
 }
 
-//------------------------------------------------
+SymbolID
+ASTVisitor::
+extractSymbolID(
+    const Decl* D)
+{
+    SymbolID id = SymbolID::zero;
+    extractSymbolID(D, id);
+    return id;
+}
 
 bool
 ASTVisitor::
 shouldSerializeInfo(
     bool PublicOnly,
-    bool IsInAnonymousNamespace,
-    NamedDecl const* D) noexcept
+    bool IsOrIsInAnonymousNamespace,
```

Comment:
Is Or Is ?

Reply:
"Is" or "IsIn"

Reply:
What does "Is Or Is In" mean

Reply:
_is_ this declaration an anonymous namespace or _is_ this declaration _in_ an anonymous namespace?